src/gt4py/gtc/python/npir.py

```diff
@@ -1,12 +1,22 @@
-from typing import List, Tuple
+from typing import List, Tuple, cast
 
 import eve
 from pydantic import validator
 
-from gt4py.gtc import common, gtir
+from gt4py.gtc import common
 
 
-class Literal(gtir.Literal):
+class Expr(common.Expr):
+    # MARK: remove when abstract nodes implemented in eve
+    def __init__(self, *args, **kwargs):
+        if type(self) is Expr:
+            raise TypeError("Cannot instantiate abstract Expr type of numpy IR")
+        super().__init__(*args, **kwargs)
+
+
+class Literal(common.Literal, Expr):
+    kind = cast(common.ExprKind, common.ExprKind.SCALAR)
```

Comment:
What's this cast doing?
Compared to 
```
    kind: common.ExprKind = common.ExprKind.SCALAR
```

Reply:
It simply tells mypy that this is an ExprKind instance, not an int (it gets confused for some reason). At runtime it does nothing.